unexpected EOF (again)

Using packer `v1.5.6`

```
==> Some builds didn't complete successfully and had errors:
--> ami: 1 error(s) occurred:

* Post-processor failed: unexpected EOF

==> Builds finished but no artifacts were created.
2020/06/16 06:48:47 packer-post-processor-amazon-ami-management plugin: panic: runtime error: invalid memory address or nil pointer dereference
2020/06/16 06:48:47 packer-post-processor-amazon-ami-management plugin: [signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x1d397a2]
2020/06/16 06:48:47 packer-post-processor-amazon-ami-management plugin:
2020/06/16 06:48:47 packer-post-processor-amazon-ami-management plugin: goroutine 117 [running]:
2020/06/16 06:48:47 packer-post-processor-amazon-ami-management plugin: main.(*Cleaner).DeleteImage(0xc0000b46c0, 0xc0001ea600, 0x0, 0x1)
2020/06/16 06:48:47 packer-post-processor-amazon-ami-management plugin: 	/home/runner/work/packer-post-processor-amazon-ami-management/packer-post-processor-amazon-ami-management/cleaner.go:140 +0x212
2020/06/16 06:48:47 packer-post-processor-amazon-ami-management plugin: main.(*PostProcessor).PostProcess(0xc00031c800, 0x24bab20, 0xc000478b80, 0x24c1300, 0xc0004ac7e0, 0x24c1360, 0xc0004a4ae0, 0x1, 0x203000, 0x28, ...)
2020/06/16 06:48:47 packer-post-processor-amazon-ami-management plugin: 	/home/runner/work/packer-post-processor-amazon-ami-management/packer-post-processor-amazon-ami-management/post-processor.go:100 +0x34e
2020/06/16 06:48:47 packer-post-processor-amazon-ami-management plugin: github.com/hashicorp/packer/packer/rpc.(*PostProcessorServer).PostProcess(0xc0001e37c0, 0x1, 0xc0004d2010, 0x0, 0x0)
2020/06/16 06:48:47 packer-post-processor-amazon-ami-management plugin: 	/home/runner/go/pkg/mod/github.com/hashicorp/packer@v1.5.4/packer/rpc/post_processor.go:109 +0x277
2020/06/16 06:48:47 packer-post-processor-amazon-ami-management plugin: reflect.Value.call(0xc000045da0, 0xc00000eb60, 0x13, 0x21ccdc1, 0x4, 0xc0004c4f18, 0x3, 0x3, 0x0, 0x2f9ce00, ...)
2020/06/16 06:48:47 packer-post-processor-amazon-ami-management plugin: 	/opt/hostedtoolcache/go/1.13.8/x64/src/reflect/value.go:460 +0x5f6
2020/06/16 06:48:47 packer-post-processor-amazon-ami-management plugin: reflect.Value.Call(0xc000045da0, 0xc00000eb60, 0x13, 0xc00047e718, 0x3, 0x3, 0xc0004cc000, 0xc0004be030, 0xc)
2020/06/16 06:48:47 packer-post-processor-amazon-ami-management plugin: 	/opt/hostedtoolcache/go/1.13.8/x64/src/reflect/value.go:321 +0xb4
2020/06/16 06:48:47 packer-post-processor-amazon-ami-management plugin: net/rpc.(*service).call(0xc0001e3800, 0xc0000ab400, 0xc00036e9a0, 0xc00036e9b0, 0xc0000fca80, 0xc00031f9a0, 0x1e5bfe0, 0xc0004be050, 0x18a, 0x1e091c0, ...)
2020/06/16 06:48:47 packer-post-processor-amazon-ami-management plugin: 	/opt/hostedtoolcache/go/1.13.8/x64/src/net/rpc/server.go:377 +0x16f
2020/06/16 06:48:47 packer-post-processor-amazon-ami-management plugin: created by net/rpc.(*Server).ServeCodec
2020/06/16 06:48:47 packer-post-processor-amazon-ami-management plugin: 	/opt/hostedtoolcache/go/1.13.8/x64/src/net/rpc/server.go:474 +0x42b
2020/06/16 06:48:47 [INFO] (telemetry) ending amazon-ami-management
2020/06/16 06:48:47 [INFO] (telemetry) Starting post-processor manifest
2020/06/16 06:48:47 /Users/lbriggs/.packer.d/plugins/packer-post-processor-amazon-ami-management: plugin process exited
2020/06/16 06:48:47 [INFO] (telemetry) ending manifest
2020/06/16 06:48:47 Flagging to keep original artifact from post-processor 'manifest'

* Post-processor failed: unexpected EOF
2020/06/16 06:48:47 machine readable: error-count []string{"1"}
==> Some builds didn't complete successfully and had errors:
2020/06/16 06:48:47 machine readable: ami,error []string{"1 error(s) occurred:\n\n* Post-processor failed: unexpected EOF"}

* Post-processor failed: unexpected EOF
==> Builds finished but no artifacts were created.
2020/06/16 06:48:47 [INFO] (telemetry) Finalizing.
2020/06/16 06:48:47 waiting for all plugin processes to complete...
2020/06/16 06:48:47 /usr/local/bin/packer: plugin process exited
2020/06/16 06:48:47 /usr/local/bin/packer: plugin process exited
2020/06/16 06:48:47 /usr/local/bin/packer: plugin process exited
2020/06/16 06:48:47 /usr/local/bin/packer: plugin process exited
make: *** [build] Error 1
```

--- a/README.md
+++ b/README.md
@@ -67,8 +67,8 @@ source "amazon-ebs" "example" {
   instance_type = "t2.micro"
   ssh_username = "ec2-user"
   ssh_pty = true
-  ami_name = "packer-example"
-  tags {
+  ami_name = "packer-example ${formatdate("YYYYMMDDhhmmss", timestamp())}"
+  tags = {
     Amazon_AMI_Management_Identifier = "packer-example"
   }
 }

--- a/cleaner.go
+++ b/cleaner.go
@@ -134,7 +134,7 @@ func (c *Cleaner) DeleteImage(image *ec2.Image) error {
 	log.Printf("Deleting snapshot related to AMI (%s)", *image.ImageId)
 	for _, device := range image.BlockDeviceMappings {
 		// skip delete if use ephemeral devise
-		if device.Ebs == nil {
+		if device.Ebs == nil || device.Ebs.SnapshotId == nil {
 			continue
 		}
 		log.Printf("Deleting snapshot (%s) related to AMI (%s)", *device.Ebs.SnapshotId, *image.ImageId)
@@ -179,6 +179,10 @@ func (c *Cleaner) setInstanceUsed() error {
 	}
 	for _, reservation := range ret.Reservations {
 		for _, instance := range reservation.Instances {
+			if instance.ImageId == nil {
+				continue
+			}
+
 			c.used[*instance.ImageId] = &Used{
 				ID:   *instance.InstanceId,
 				Type: "instance",
@@ -194,6 +198,10 @@ func (c *Cleaner) setLaunchConfigurationUsed() error {
 		return err
 	}
 	for _, lc := range ret.LaunchConfigurations {
+		if lc.ImageId == nil {
+			continue
+		}
+
 		c.used[*lc.ImageId] = &Used{
 			ID:   *lc.LaunchConfigurationName,
 			Type: "launch configuration",
@@ -217,6 +225,10 @@ func (c *Cleaner) setLaunchTemplateUsed() error {
 		}
 
 		for _, ltv := range versions.LaunchTemplateVersions {
+			if ltv.LaunchTemplateData == nil || ltv.LaunchTemplateData.ImageId == nil {
+				continue
+			}
+
 			c.used[*ltv.LaunchTemplateData.ImageId] = &Used{
 				ID:   fmt.Sprintf("%s (%d)", *ltv.LaunchTemplateName, *ltv.VersionNumber),
 				Type: "launch template",

